app/controllers/slides_controller.rb

```diff
@@ -41,7 +41,14 @@ def preview
   def draft
     @draft = @slide.find_or_create_draft
     @draft.scheduled_items.clear
-    @draft.update(slide_params)
+
+    # This will error if there are any nested scheduled items in slide_params. (See
+    # https://github.com/chapmanu/signage/issues/147.) It will look up scheduled_items using
+    # draft_id (negative) but scheduled_items are associated with proper ID.
+    # (Is this why scheduled_items are cleared above?)
+    filtered_params = slide_params.except('scheduled_items_attributes')
```

Comment:
You're right. I think any association getting updated here would cause the issue. But scheduled_items is the only one. See here:

https://github.com/chapmanu/signage/blob/b82d254133290a514c0f15e16383287fffc34826/app/controllers/slides_controller.rb#L159